azure-cli: 2.50.0 → 2.51.0

AkechiShiro opened the pull request:
## Description of changes
Bump python dependencies and also needs #246367 to be merged in order for the build to be successful.

Also adds : 
- SourceProvenance and changelog 
 
<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

AkechiShiro committed:
azure-cli: 2.50.0 → 2.51.0 https://github.com/MicrosoftDocs/azure-docs-cli/blob/main/docs-ref-conceptual/release-notes-azure-cli.md

* Add SourceProvenance and changelog
* Fix typo on sourceTypes
* Bump msal from 1.20.0 to 1.24.0b1
* Fix changelog link to latest file upstream instead of commit
* Add meta.{mainProgram,platforms,longDescription,downloadPage} and
  update sha256 to sha512

wegank commented:
@ofborg build azure-cli azure-cli.passthru.tests

AkechiShiro commented:
@wegank is there any for me to reproduce the issue locally, Ofborg uses a specific command that is more specific than a `nix-build -A azure-cli` inside the nixpkgs directory ?

I've also seen a little issue for #246367 (probably not related to the Ofborg built failure), the support for Python 3.7 was dropped, it is necessary to have Python 3.8, I will probably make another PR to fix this with the documentation that deprecated Python 3.7

I'm not sure if @jonringer will be able to review this PR, the previous one was reviewed by others reviewers and was merged.

wegank commented on pkgs/tools/admin/azure-cli/default.nix:
```diff
@@ -269,6 +269,8 @@ py.pkgs.toPythonApplication (py.pkgs.buildAzureCliPackage {
   meta = with lib; {
     homepage = "https://github.com/Azure/azure-cli";
     description = "Next generation multi-platform command line experience for Azure";
+    changelog = "https://github.com/MicrosoftDocs/azure-docs-cli/commit/a80f1a316226cf7801a25b7adb497ac13f875281";
```
```suggestion
    changelog = "https://github.com/MicrosoftDocs/azure-docs-cli/blob/main/docs-ref-conceptual/release-notes-azure-cli.md";
```

AkechiShiro replied:
Is it sure that we want to put the main branch for changelog, it will get updated automatically but release notes won't always be in sync with the version.

I thought pinning the exact commit was better, and then I'd update it at every version bump so it always stays coherent.

RaitoBezarius replied:
You didn't refer to the filename and referring to the latest changelog avoids having to change it and users can find the versions themselves.

Mic92 commented:
Result of `nixpkgs-review pr 248187` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>azure-cli</li>
    <li>azure-cli.dist</li>
  </ul>
</details>

AkechiShiro commented:
Hi @Mic92,

I've ran into this issue as well before bumping msal from 1.20.0 to 1.24.0b1, just before doing the latest commit bumping msal to 1.24.0b1 : https://github.com/AzureAD/microsoft-authentication-library-for-python/commit/cb4aa61f06787bf4b9f5bdf8464c64476e506efb#diff-fa602a8a75dc9dcc92261bac5f533c2a85e34fcceaff63b3a3a81d9acde2fc52R61
> cryptography>=0.6,<44

Was the nixpkgs-review building with the latest commit ?
I gave it a try just in case, it is building successfully on my end 
![image](https://github.com/NixOS/nixpkgs/assets/14914796/f3f84c1d-cfa6-4cb7-9cc7-f3a3c0d2b14a)

wegank commented:
Result of `nixpkgs-review pr 248187` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>azure-cli</li>
    <li>azure-cli.dist</li>
  </ul>
</details>

jonringer commented:
There seems to be a merge conflict now. Do yo mind resolving?

AkechiShiro commented:
@jonringer, rebase was redone on latest master, the conflict was coming from this : https://github.com/NixOS/nixpkgs/commit/591ccfe5b9bd6d9747c264a1aa5d324bc1cdc7dd#diff-00fd8f6e452e9a956cb6679ac2194bcdbac39442bd3670235fb6cd8b96dd3790L5-R7

I fixed it by just keeping the change that was originally made.

I added some more meta attributes as well, I think it's all good to be merged now, if anything else is needed feel free to let me know.

AkechiShiro commented:
I've just noticed something broke but due to an upstream change most likely, it seems the archive that was getting pulled is not longer working : https://hydra.nixos.org/build/232703803/nixlog/1

Is there any way to avoid this kind of breakage ? What would be an action I could take to do this ?
 
```
unpacking source archive /nix/store/psxah9s2qmnqzvhafrmdqkqizrj5vaxb-azure-mgmt-storage-21.0.0.zip
do not know how to unpack source archive /nix/store/psxah9s2qmnqzvhafrmdqkqizrj5vaxb-azure-mgmt-storage-21.0.0.zip
```
I need to open a PR to bump `azure-mgmt-storage`, latest release is : https://pypi.org/project/azure-mgmt-storage/21.1.0/
